File: Practice/LinkedList.py
class Node:
	def __init__(self, value=0):
		self.value = value
		self.next = None

class LinkedList:
	def __init__(self):
		self.head = None
		self.tail = None
		self.length = 0

	def append(self, value):
		new_node = Node(value)
		if self.length == 0:
			self.head = new_node
		else:
			self.tail.next = new_node
		self.tail = new_node
		self.length += 1

	def search(self, value):
		current = self.head
		found = False
		while not found and current != None:
			if current.value == value:
				found = True
			else:
				current = current.next
		return current

	def insert(self, value, searchVal):
		new_node = Node(value)
		currentNode = self.search(searchVal)
		if currentNode == None:
			return "Value not found"
		else:
			new_node.next = currentNode.next
			currentNode.next = new_node
			self.length += 1

			if new_node.next == None:
				self.tail = new_node

	def contain(self, value):
		currentNode = self.search(value)
		if currentNode == None:
			return False
		else:
			return True

	def delete(self, value):
		current = self.head
		previous = None
		found = False
		while not found and current != None:
			if current.value == value:
				found = True
			else:
				previous = current
				current = current.next

		if previous == None:
			self.head = current.next
			current.next = None
		elif current.next == None:
			self.tail = previous
			previous.next = None
		else:
			previous.next = current.next
			current.next = None
		

		self.length -= 1

File: Linked List/Linked List.js
function Node(value){
	this.value = value;
	this.next = null;
}

function LinkedList(){
	this.head = null;
	this.tail = null;
	this.size = 0;
}

LinkedList.prototype.push = function(value){
	var currentNode = this.head;
	var newNode = new Node(value);
	if(this.size === 0){
		this.head = newNode;
	} else{
		while(currentNode.next != null){
			currentNode = currentNode.next;
		}
		currentNode.next = newNode;
	}
	this.tail = newNode;
	this.size++;
}

LinkedList.prototype.delete = function(value){
	var currentNode = this.head;
	var previousNode;
	var found = false;
	if(value == this.head.value){
		this.head = this.head.next;
		this.size--;
	} else{
		while(found === false && currentNode !== null){
			if(value === currentNode.value){
				found = true;
			}
			else{
				previousNode = currentNode;
				currentNode = currentNode.next;
			}
		}
		if(found === true){
			if(currentNode.next === null){
				previousNode.next = null;
				this.tail = previousNode;
			} else{
				previousNode.next = currentNode.next;
				currentNode.next = null;
			}
			this.size--;
		}
	}
}

var ll = new LinkedList();
ll.push(1);
ll.push(2);
ll.push(3);
ll.push(4);

console.log(ll.head);
console.log(ll.tail);
console.log(ll.size);